### ISSUE TITLE
Locale codes mismatch between Phrase OTA and Android

### ISSUE THREAD

**Issue by emiltsang86** [phrase/phrase-android#41]
**Locales in Phrase**    |    **Locales in Android, Locales found in requests**
en | en_HK
ja | ja_JP
ko | ko_KR
zh-Hans | zh_CN_#Hans
zh-Hant | zh_HK_#Hant

When I try to request the `translation xml` with the link containing the locale value `zh_HK_#Hant`,
`https://ota.phraseapp.com/DISTRIBUTION_ID/DEV_SECRET/zh_HK_#Hant/xml?`,
a 404 Not Found error is returned.

After checking with `Phrase support`,
they found out the locale codes in the requests do not match the codes using in the project.

It seems Android added the `location` to the locale code, i.e. `_HK` in `en_HK` .
For `English`, `Japanese`, `Korean`, Phrase OTA can get the `translation xml` correctly even the location is included in the locale code, 
i.e. Phrase OTA can handle to redirection from the `url` with `en_HK`,  `ja_JP`, `ko_KR`
to `translation xml` for `English`, `Japanese`, `Korean`.
It ONLY failed for Simplified Chinese and Traditional Chinese.

Then I have called the `Phrase.setLocaleCode` to update the locale to `zh-Hans` / `zh-Hant`,
enforcing `Phrase OTA` to get the correct `translation xml` for Simplified Chinese / Traditional Chinese.

However, I found that after calling `Phease.setLocaleCode` with `zh-Hans` / `zh-Hant`,
some UI components cannot be completely translated,
see the `Day of Week` row, `Cancel` / `OK` buttons for `DatePicker`,
which they should be displayed in `Chinese` instead of `English`.
<img width="274" alt="image" src="https://user-images.githubusercontent.com/104979030/166956397-e4cf738a-2752-4ae8-9d93-654f06c433c6.png">

In short, I have to set the locale with Phrase.setLocaleCode("zh-Hant") to fetch the translation xml.
However, `Phrase.setLocaleCode(“zh-Hant”)` may change the locale settings in `Android context` which broke the localization in DatePicker.

**Comment by theSoenke** [comment]
Hi @emiltsang86, thanks for your report. We have just published a new release 3.1.3, which will convert the Android locale codes into the Phrase locale codes https://github.com/phrase/phrase-android/releases/tag/3.1.3

When you have the option `Use language without region as fallback` for the distribution enabled it will also use `zh-Hans` as the fallback when for example requesting translations for `zh-Hans-CN`.

**Comment by emiltsang86** [comment]
Hi @theSoenke ,
I have found `another case` of Locale codes mismatch between Phrase OTA and Android.
Due to the mismatch,
the translation file cannot be fetched in `Android 6`,
and translation remains as default language `English`.

In `Android 6`, the locale code presentation is different from `Android 7`,
which does NOT contain `_Hant` for Chinese Traditional,
`_Hans` for Chinese Simplified.

**Locales in Phrase | Locales in Android, Locales found in requests in Android 6**
zh-Hans | zh_CN
zh-Hant | zh_TW

Url when requesting translation files in `Android 6`
`Chinese Traditional`: https://ota.phraseapp.com/DISTRIBUTION_ID/DEV_SECRET/`zh_TW`/xml?
`Chinese Simplified`: https://ota.phraseapp.com/DISTRIBUTION_ID/DEV_SECRET/`zh_CN`/xml?

**Comment by theSoenke** [comment]
Hey @emiltsang86, I think this case is a bit more tricky to handle because `zh_TW` and `zh_CN` are valid locale codes. So I don't think we can automatically convert them. What I would suggest instead to create the locales `zh_TW` and `zh_CN` in Phrase. And instead of translating them I would suggest to set a fallback locale to e.g. `zh-Hans` in the locale settings:
![Screenshot from 2022-08-03 16-27-27](https://user-images.githubusercontent.com/1757301/182633698-11480715-b287-4b0e-8835-b553a311c386.png)

In addition "use fallback language" needs to be enabled in the distribution. Afterward, the request for those locales will return the fallback locale. Would this be a solution?

**Comment by RogerParis** [comment]
@emiltsang86 it would be interesting to know if this can work for you.

**Comment by theSoenke** [comment]
Hi @emiltsang86, I would close this issue for now but let me know if you should still experience any issues with the locales
